### PR TITLE
Move references to separate XSL file

### PR DESCRIPTION
**Pull request by redhatrises**
- Makes it easier to update reference mapping rather than searching for them
- Allows references to also be consumed by xccdf-guide-impl.xsl

### CODE REVIEW

**Comment by redhatrises** [comment]
@openscap-ci test this please

**Comment by redhatrises** [comment]
Needed for #983

File: xsl/xccdf-references.xsl [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+     That is fine because we will go through just the benchmark references
+     and only then we compare to this map. So this is correct.
+     See template "get-all-references". -->
+<xsl:key name="references" match="//cdf:reference" use="@href"/>
+
+<xsl:output
+    method="html"
+    encoding="utf-8"
+    indent="no"
+    omit-xml-declaration="yes"/>
```
Comment (mpreisler) [inline xsl/xccdf-references.xsl]: this probably shouldn't be here

**Comment by redhatrises** [comment]
@mpreisler fixed

**Comment by mpreisler** [comment]
ACK

**Pull request merged**